core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/EnvelopeSimPath.java

```diff
@@ -135,25 +136,58 @@ public void setElectricalProfiles(Map<String, RangeMap<Double, String>> electric
 
         for (var powerClassEntry : electricalProfilesByPowerClass.entrySet()) {
             var profileMap = mergeRanges(powerClassEntry.getValue());
-            TreeRangeMap<Double, ModeAndProfile> profilesAndModes = TreeRangeMap.create();
-            profilesAndModes.put(Range.closed(0.0, length), new ModeAndProfile(null, null));
+            TreeRangeMap<Double, ElecCond> profilesAndModes = TreeRangeMap.create();
             profilesAndModes.putAll(catenaryModeMap);
 
             for (var profileEntry : profileMap.asMapOfRanges().entrySet()) {
                 for (var entry : catenaryModeMap.subRangeMap(profileEntry.getKey()).asMapOfRanges().entrySet()) {
                     var mode = entry.getValue().mode();
                     var profileLevel = profileEntry.getValue();
-                    profilesAndModes.put(entry.getKey(), new ModeAndProfile(mode, profileLevel));
+                    profilesAndModes.put(entry.getKey(), new ElecCond(mode, profileLevel, null));
                 }
             }
             modeAndProfileMapsByPowerClass.put(powerClassEntry.getKey(), ImmutableRangeMap.copyOf(profilesAndModes));
         }
     }
 
-    public RangeMap<Double, ModeAndProfile> getModeAndProfileMap(String powerClass) {
-        if (modeAndProfileMapsByPowerClass == null)
-            return catenaryModeMap;
-        return modeAndProfileMapsByPowerClass.getOrDefault(powerClass, catenaryModeMap);
+    private RangeMap<Double, ElecCond> getModeAndProfileMap(String powerClass, Range<Double> range,
+                                                           boolean ignoreElectricalProfiles) {
+        if (modeAndProfileMapsByPowerClass == null || ignoreElectricalProfiles)
+            return catenaryModeMap.subRangeMap(range);
+        return modeAndProfileMapsByPowerClass.getOrDefault(powerClass, catenaryModeMap).subRangeMap(range);
+    }
+
+    /**
+     * Get the catenary related data for a given power class and power restriction map.
+     */
+    public ImmutableRangeMap<Double, ElecCond> getElecCondMap(String basePowerClass,
+                                                     RangeMap<Double, String> powerRestrictionMap,
+                                                     Map<String, String> powerRestrictionToPowerClass,
+                                                     boolean ignoreElectricalProfiles) {
```

Comment:
With these new methods, the `EnvelopeSimPath` is diverging from its abstract interface `PhysicsPath`, which means we need to use the implementation type in interfaces.

Would it make sense to add all the public methods to `PhysicsPath` and replace `EnvelopeSimPath` with `PhysicsPath` in any place that isn't the instantiation?

Reply:
Indeed, they are diverging, however `EnvelopeSimPath` is used for typing interfaces only outside of the `envelope-sim` module. Inside said module only `PhysicsPath` is used for typing interfaces.
So I wouldn't say it's okay if they diverge ? What do you think @multun ?

I would say that we could move EnvelopeSimPath out of the module (and also rename it), since it's only used outside of the module anyway

Reply:
`PhysicsPath` (and `envelope-sim`) should only concern itself with physics simulation I guess. It's okay to have more featureful implementations in more generalist modules.

Maybe we could carve more restricted interfaces for other purpose, but I don't think this is the point of this PR